src/ImageSharp/Formats/Jpeg/JpegEncoder.cs

```diff
@@ -50,7 +56,21 @@ public Task EncodeAsync<TPixel>(Image<TPixel> image, Stream stream, Cancellation
             where TPixel : unmanaged, IPixel<TPixel>
         {
             var encoder = new JpegEncoderCore(this);
+            this.EnrichColorType<TPixel>();
             return encoder.EncodeAsync(image, stream, cancellationToken);
         }
+
+        /// <summary>
+        /// If ColorType was not set, set it based on the given <typeparamref name="TPixel"/>.
+        /// </summary>
+        private void EnrichColorType<TPixel>()
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            if (this.ColorType == null)
+            {
+                bool isGrayscale = typeof(TPixel) == typeof(L8) || typeof(TPixel) == typeof(L16);
```

Comment:
We should be storing the `JpegColorType` in the `JpegMetadata` class upon decode and querying both places with the encoder options taking precedence. 

`La16` and `La32` should trigger grayscale also.

Reply:
Added La16 and La32 to JpegEncoder and added test cases for both.

Reply:
Unresolving the conversation, since I think @JimBobSquarePants also asked for a decoder extension here, will try to explain  what he likely meant in more detailed manner:
- Define an optional `JpegColorType?` property in `JpegMetadata`
- Detect `JpegColorType` in `JpegDecoderCore` and store it into the image's `JpegMetadata`.
- In the encoder, also trigger the grayscale path in case if  `JpegMetadata.JpegColorType` is set to `JpegColorType.GrayScale`

@ynse01 let us know if you feel comfortable and confident adding this. If not, I recommend to not block the PR on this,  and open a tracking issue for the addition.

Reply:
I gave it a try. Please indicate if this is what you guys expect.